yoyodyne/predict.py

```diff
@@ -66,119 +155,51 @@ def write_predictions(
-    util.log_info("Prediction complete")
-
-
-@click.command()
-@click.option("--experiment", required=True)
-@click.option("--predict", required=True)
-@click.option("--output", required=True)
-@click.option("--model-dir", required=True)
-@click.option("--checkpoint", required=True)
-@click.option(
-    "--source-col", type=int, default=1, help="1-based index for source column"
-)
-@click.option(
-    "--target-col", type=int, default=2, help="1-based index for target column"
-)
-@click.option(
-    "--features-col",
-    type=int,
-    default=0,
-    help="1-based index for features column; "
-    "0 indicates the model will not use features",
-)
-@click.option("--source-sep", type=str, default="")
-@click.option("--target-sep", type=str, default="")
-@click.option("--features-sep", type=str, default=";")
-@click.option("--tied-vocabulary", default=True)
-@click.option(
-    "--arch",
-    type=click.Choice(
-        [
-            "feature_invariant_transformer",
-            "lstm",
-            "pointer_generator_lstm",
-            "transducer",
-            "transformer",
-        ]
-    ),
-    required=True,
-)
-@click.option("--batch-size", type=int, default=1)
-@click.option(
-    "--beam-width", type=int, help="If specified, beam search is used"
-)
-@click.option(
-    "--attention/--no-attention",
-    type=bool,
-    default=True,
-    help="Use attention (`lstm` only)",
-)
-@click.option("--bidirectional/--no-bidirectional", type=bool, default=True)
-@click.option("--accelerator")
-def main(
-    experiment,
-    predict,
-    output,
-    model_dir,
-    checkpoint,
-    source_col,
-    target_col,
-    features_col,
-    source_sep,
-    target_sep,
-    features_sep,
-    tied_vocabulary,
-    arch,
-    batch_size,
-    beam_width,
-    attention,
-    bidirectional,
-    accelerator,
-):
+
+
+def main() -> None:
     """Predictor."""
-    config = dataconfig.DataConfig(
-        source_col=source_col,
-        features_col=features_col,
-        target_col=target_col,
-        source_sep=source_sep,
-        features_sep=features_sep,
-        target_sep=target_sep,
-        tied_vocabulary=tied_vocabulary,
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "--experiment", required=True, help="Name of experiment"
     )
-    # TODO: Do not need to enforce once we have batch beam decoding.
-    if beam_width is not None:
-        util.log_info("Decoding with beam search; forcing batch size to 1")
-        batch_size = 1
-    dataset_cls = datasets.get_dataset_cls(config.has_features)
-    dataset = dataset_cls(predict, config)
-    dataset.load_index(model_dir, experiment)
-    util.log_info(f"Source vocabulary: {dataset.source_symbol2i}")
-    util.log_info(f"Target vocabulary: {dataset.target_symbol2i}")
-    collator_cls = collators.get_collator_cls(
-        arch, config.has_features, include_targets=False
+    # Path arguments.
+    parser.add_argument(
+        "--predict",
```

Comment:
Can this be called `--predict_path` or `--prediction_path`? `--predict` sounds like it'd be a boolean flag to me :D

Reply:
Well, the tool is called predict so I figure... but okay, I can refactor around that.